spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -425,14 +431,70 @@ class DLModel[@specialized(Float, Double) T: ClassTag](
   }
 
   override def copy(extra: ParamMap): DLModel[T] = {
-    val copied = new DLModel(model, featureSize, uid).setParent(parent)
+    val copied = new DLModel(_model, featureSize, uid).setParent(parent)
     copyValues(copied, extra)
   }
+
+  private var _model: Module[T] = model
+
+  override protected def getModule: Module[T] = _model
+
+  override protected def setModule(module: Module[T]): Unit = _model = module
 }
 
-// TODO, add save/load
 object DLModel {
+  def load[@specialized(Float, Double) T: ClassTag]
+  (path: String)(implicit ev: TensorNumeric[T]): DLModel[T] = {
+    DLModelBase.load[T, DLModel[T]](path).asInstanceOf[DLModel[T]]
+  }
+}
+
+/**
+ * Base class of DLModels, such as DLClassifierModel, provide (pipeline) model IO features.
+ * Please make sure working with module which returned by [[getModule]] and set by [[setModule]].
+ */
+abstract class DLModelBase[@specialized(Float, Double) T: ClassTag, M <: DLTransformerBase[M]]
```

Comment:
Is it possible to merge this into DLModel?

Reply:
I think DLModelBase can be extend by other model like transformer based on a bigdl.Module which need IO supporting.  
Actually, I am working on DAGEstimator/DAGModel which supporting feature engineering preprocess before training。

Reply:
That's separate topic. Still I would vote to put the save/load code in DLModel for now, and we can migrate it to DLModelBase in your future PR.
As long as we keep the API stable, it will not impact user experience.